cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -512,30 +562,25 @@ where
         Ok(())
     }
 
-    /// Check if the given store paths _exists_ on the filesystem,
-    /// or in the configured nix store.
+    /// Check if the given store paths exists in the configured nix store.
     /// Substitute store paths if necessary and possible.
+    /// Sets a gc root in [Self::gc_root_base_path] on success.
     ///
-    /// If the store paths do not exist,
-    /// the function will fall back to querying the nix store for the store paths.
-    /// Formerly, this function checked the store paths with `nix build` immediately,
-    /// which would also ensure the integrity of the references of the store paths.
-    /// However, the runtime profile of the `nix build` command
-    /// has significant overhead for large environments.
-    /// 50ms to 100ms per package in an environment of 50 packages,
-    /// is very noticeable.
-    /// To address this we replace the nix call with a number of `stat`
-    /// calls for the paths that are checked, with the optimistic assumption
-    /// that if a path exists, it and its references are valid.
-    /// If they are not, we fall back to the nix call,
-    /// which checking against alternative stores and substitution from binary caches.
+    /// This methods is expected to be called _for all outputs of a derivation_
+    /// iff any output of the derivation has formerly been identified as invalid
+    /// by [Self::pre_check_store_paths].
+    ///
+    /// To avoid GC of substituted paths, pass an `--out-links` argument,
+    /// to create "temproots".
     fn check_store_path_with_substituters(
         &self,
+        install_id: &str,
         paths: impl IntoIterator<Item = impl AsRef<OsStr>>,
     ) -> Result<bool, BuildEnvError> {
         let mut cmd = self.base_command();
         cmd.arg("build");
-        cmd.arg("--no-link");
+        cmd.arg("--out-link");
+        cmd.arg(self.new_gc_root_path(format!("by-store-path-{install_id}")));
```

Comment:
blocking question: if this gets called for multi-output install ids will there be a path collision?